File: dotnet/src/Downstairs.Infrastructure/Persistence/Models/CreditCreditTransaction.cs
namespace Downstairs.Infrastructure.Persistence.Models;

public partial class CreditCreditTransaction
{
    public long Id { get; set; }

    public long CreditId { get; set; }

    public long CreditTransactionId { get; set; }

    public byte Amount { get; set; }

    public virtual Credit Credit { get; set; } = null!;

    public virtual CreditTransaction CreditTransaction { get; set; } = null!;
}

File: php-laraval-app/app/Models/CreditCreditTransaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CreditCreditTransaction extends Model
{
    protected $table = 'credit_credit_transaction';

    /**
     * Indicates if the model should be timestamped.
     *
     * @var bool
     */
    public $timestamps = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'credit_id',
        'credit_transaction_id',
        'amount',
    ];

    public function credit(): BelongsTo
    {
        return $this->belongsTo(Credit::class);
    }

    public function creditTransaction(): BelongsTo
    {
        return $this->belongsTo(CreditTransaction::class);
    }
}
